### PR TITLE
[SHIRO-811] fix toggling test through Mockito conversion.

### PR DESCRIPTION
**Pull request by bmarwell**
This PR converts tests from the web project from using EasyMock into Mockito. 
The idea is that EasyMock creates mocks which seem unpredictable in a few cases we encounter, leading to flaky tests.

https://issues.apache.org/jira/browse/SHIRO-811

---

Following this checklist to help us incorporate your contribution quickly and easily:

 - [X] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SHIRO) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [X] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[SHIRO-XXX] - Fixes bug in SessionManager`,
       where you replace `SHIRO-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the commit message.
 - [X] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [X] Run `mvn clean install apache-rat:check` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [X] If you have a group of commits related to the same change, please squash your commits into one and force push your branch using `git rebase -i`. 
 
Trivial changes like typos do not require a JIRA issue (javadoc, comments...). 
In this case, just format the pull request title like `(DOC) - Add javadoc in SessionManager`.
 
If this is your first contribution, you have to read the [Contribution Guidelines](https://github.com/apache/shiro/blob/master/CONTRIBUTING.md)

If your pull request is about ~20 lines of code you don't need to sign an [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) 
if you are unsure please ask on the developers list.

To make clear that you license your contribution under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [X] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
 - [X] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

**Comment by bmarwell** [comment]
Still fails: https://ci-builds.apache.org/job/Shiro/job/Shiro-all/view/change-requests/job/PR-282/lastCompletedBuild/testReport/

**Comment by bmarwell** [comment]
Huh. Maybe there is still some Powermock stuff going around in the background. Wouldn't be the first time we had issues with it. I'll look for other occurrences.

File: web/src/test/java/org/apache/shiro/web/filter/authc/BasicHttpFilterAuthenticationTest.java
Comment (bdemers) [inline web/src/test/java/org/apache/shiro/web/filter/authc/BasicHttpFilterAuthenticationTest.java:177]: Good call on these!

**Review by bdemers — APPROVED** [review]
🚢

**Comment by bmarwell** [comment]
Hi Brian,
I will not merge this PR as long as there are those toggling test results: https://ci-builds.apache.org/job/Shiro/job/Shiro-all/view/change-requests/job/PR-282/

I just need to have a non-draft PR so I can see the Jenkins output.

By the way, the test never failed on my machine. I cannot even reproduce it.

**Comment by bmarwell** [comment]
Hi @bdemers,
feel free to re-review. I converted some more tests, and now the "flaky test" seems to have gone for good.

**Review by bdemers — APPROVED** [review]

**Pull request merged**
